src/services/block.service.ts

```diff
@@ -23,6 +24,30 @@ export class BlockService {
     return await this.blockStorage.getMinedBlockByHash(hash);
   }
 
+  public async getBlockStats(): Promise<any> {
+    const allBlocks = await this.blockStorage.getAllBlocks();
+    let averageBlockTime = 0;
+    let txThroughput = 0;
+    if (allBlocks && allBlocks.length > 1) {
+      allBlocks.sort((a, b) => b.timestamp - a.timestamp);
+      const timeDiff = (allBlocks[0].timestamp - allBlocks[allBlocks.length - 1].timestamp) / 1000;
+      averageBlockTime = allBlocks.length / timeDiff;
+
+      const totalNumOfTxs = _.reduce(
+        allBlocks,
+        (prev, curr) => {
+          return prev + curr.txs.length;
+        },
+        0,
+      );
+      txThroughput = totalNumOfTxs / (timeDiff / 1000);
```

Comment:
doesn't need to divide by 1000, it's done before

Reply:
ah, yes. so stupid haha